MerlinAU.sh

```diff
@@ -1810,58 +1810,55 @@ _DoCleanUp_()
    fi
 }
 
-##------------------------------------------##
-## Modified by ExtremeFiretop [2024-Mar-15] ##
-##------------------------------------------##
-check_memory_and_prompt_reboot() {
+##----------------------------------------##
+## Modified by Martinski W. [2024-Mar-16] ##
+##----------------------------------------##
+check_memory_and_prompt_reboot()
+{
     local required_space_kb="$1"
     local availableRAM_kb="$2"
 
-    if [ "$availableRAM_kb" -lt "$required_space_kb" ]; then
+    if [ "$availableRAM_kb" -lt "$required_space_kb" ]
+    then
         Say "Insufficient RAM available."
 
         # Attempt to clear PageCache #
         Say "Attempting to free up memory..."
         sync; echo 1 > /proc/sys/vm/drop_caches
+        sleep 2
 
         # Check available memory again #
-        availableRAM_kb=$(_GetAvailableRAM_KB_)
-        if [ "$availableRAM_kb" -lt "$required_space_kb" ]; then
-
+        availableRAM_kb="$(_GetAvailableRAM_KB_)"
+        if [ "$availableRAM_kb" -lt "$required_space_kb" ]
+        then
             freeRAM_kb="$(get_free_ram)"
             Say "Required RAM: ${required_space_kb} KB - RAM Free: ${freeRAM_kb} KB - RAM Available: ${availableRAM_kb} KB"
             
             # Attempt to clear dentries and inodes. #
             Say "Attempting to free up memory again more aggressively..."
-
             sync; echo 2 > /proc/sys/vm/drop_caches
+            sleep 2
 
             # Check available memory again #
-            availableRAM_kb=$(_GetAvailableRAM_KB_)
-            if [ "$availableRAM_kb" -lt "$required_space_kb" ]; then
-
+            availableRAM_kb="$(_GetAvailableRAM_KB_)"
+            if [ "$availableRAM_kb" -lt "$required_space_kb" ]
+            then
                 freeRAM_kb="$(get_free_ram)"
                 Say "Required RAM: ${required_space_kb} KB - RAM Free: ${freeRAM_kb} KB - RAM Available: ${availableRAM_kb} KB"
                 
                 # Attempt to clear clears pagecache, dentries, and inodes after shutting down services		
                 Say "Attempting to free up memory once more even more aggressively..."
 
-                # Check if '/opt/bin/diversion' exists #
-                if [ -f /opt/bin/diversion ]; then
-                    # Stop Diversion services before flash #
-                    Say "Stopping Diversion service..."
-                    /opt/bin/diversion unmount
-                fi
-
-                # Stop entware services before F/W flash #
+                # Stop Entware services before F/W flash #
                 _EntwareServicesHandler_ stop
 
                 sync; echo 3 > /proc/sys/vm/drop_caches
+                sleep 2
```

Comment:
Smart call, this would be the heaviest clear and also with the function to stop the services giving it a few seconds to settle isn't a bad idea. Sometimes forget that not everyone has ROG router with all the resources. lol